### ISSUE TITLE
Clearance

### ISSUE THREAD

**Issue by deece** [EvilHacker/BoxJoint#3]
I'd like to have a clearance factor for an easier fit.

I thought the margin parameter would accomplish this, but setting a large margin doesn't alter the clearance between the tab and slot.

![image](https://github.com/EvilHacker/BoxJoint/assets/1862055/a9aff812-8778-45d1-b9ec-619f400ae211)
![margin](https://github.com/EvilHacker/BoxJoint/assets/1862055/3a439bfe-a304-404f-9569-39ca63388697)

**Comment by Gregrvr** [comment]
I agree, having a clearance tolerance would be a great feature. I used his fusion manufacturing model and turned each of the operations into a template and then copied those templates into my setup. I added -0.003 in stock to leave in each of the 3 operations. Probably could have gotten away with -0.002 using 1/2" plywood for a 20" x 20" drawer. It made all edges 0.006" smaller but made assembly much easier.

![2024-03-17 16 10 35](https://github.com/EvilHacker/BoxJoint/assets/123017250/7a718fbf-2918-4b65-bb96-92d427778f7e)

I used 1/4" bit in my design but cut the parts using a 3/16" 2 flute down cut endmill to avoid burrying the 1/4" bit into the inside radii. My setup was 0.004 feed per tooth at 20,000 rpm and the finish was great. This is a picture of the unsanded dry-fit. Panels hand assembled without a mallet but no slop, perfect for gluing together.

![2024-03-17 16 17 17](https://github.com/EvilHacker/BoxJoint/assets/123017250/7056b1ff-5595-453a-b676-a886ab260740)

Great job on the add-in! A blind Box joint would be much faster to machine using a T-Bone joint. The outside fillets take a ton of time to machine for very little benefit!! I wish I knew python to program this style of joint! Is this something you would be willing to tackle?

![Full blind box joint](https://github.com/EvilHacker/BoxJoint/assets/123017250/4c4396a0-2c61-4088-8992-3e835b456e72)

**Comment by holla2040** [comment]
I've been this add-in https://github.com/FlorianPommerening/FingerJoints for a while. It works and is very flexible except it doesn't edit like a timeline feature. First thing I checked for here was clearance. This is definitely needed for glue up. Otherwise this looks awesome!

**Comment by it-ony** [comment]
@holla2040 a while ago, I was modifying the FingerJoints plugin to support updates. See my [PR](https://github.com/FlorianPommerening/FingerJoints/pull/26). It doesn't use a custom feature, so it doesn't work as reliable. 

Now that I discovered boxjoint, I can checkout the code, what it does differently to have a custom feature and basically combine both concepts.

**Comment by th0mpy** [comment]
I'd love to see the margins work as well. +1 for me!

**Comment by tperryba** [comment]
Margins are not clearance, i was looking at the code and playing with them they are setbacks from the the two sides, ie if you have  a 10" board and set margins to 3", you will only have fingers in the middle 4".      I do wish there was a clearance and have been looking at how to add it, but dang is this a bit mind bending on the fusion api and geometry, it is a slow process to figure out how this all works.

**Comment by Jens5395** [comment]
Great add-in Mark - thanks!
Will you be adding a 'Clearance' parameter?

**Comment by EvilHacker** [comment]
Clearance parameters have been added in since v1.1.0 of this add-in.
